packages/peregrine/lib/util/findAllMatchingVariants.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * TODO Document
+ */
+export const findAllMatchingVariants = ({
+    variants,
+    optionCodes,
+    singleOptionSelection
+}) => {
+    return variants.filter(({ attributes, product }) => {
+        const customAttributes = (attributes || []).reduce(
+            (map, { code, value_index }) => new Map(map).set(code, value_index),
+            new Map()
+        );
+        for (const [id, value] of singleOptionSelection) {
+            const code = optionCodes.get(id);
+            //if (!value) return false;
```

Comment:
Lets remove these commented out pieces of code if we're not going to include it in the final implementation